k4e-agent-sti.sh

```diff
@@ -0,0 +1,106 @@
+#!/bin/bash
+
+set -e
+
+# The scripts performs the following steps on rhel-8.5:
+# * Install imagebuilder 
+# * Create rpms for yggdrasil and k4e-device-worker
+# * Publish rpms as yum repo to be consumed by imagebuilder
+# * Publish aarch64 rhel-image for edge-device
+
+# The script assumes the machine is registered via subscription-manager
+if ! subscription-manager refresh; then
+   echo  "The machine is not registered."
+   exit 1
+fi
+
+#---------------------------
+# Install osbuild components
+#---------------------------
+dnf install -y osbuild-composer composer-cli cockpit-composer bash-completion
+systemctl enable osbuild-composer.socket --now
+systemctl enable cockpit.socket --now
+systemctl start firewalld
+firewall-cmd --add-service=cockpit && firewall-cmd --add-service=cockpit --permanent
+source  /etc/bash_completion.d/composer-cli
+
+#-----------------------------------
+# Build packages for k4e from source
+#-----------------------------------
+mkdir /home/builder
+
+dnf install -y dbus-devel systemd-devel git golang rpm-build
+rm -rf ~/rpmbuild/rpmbuild/*
+
+# Build yggdrasil rpm
+git clone https://github.com/jakub-dzon/yggdrasil.git /home/builder/yggdrasil
+cd /home/builder/yggdrasil
+export CGO_ENABLED=0
+
+export ARCH=$(uname -i)
+GOPROXY=proxy.golang.org,direct PWD=$PWD spec=$PWD outdir=$PWD make -f .copr/Makefile srpm
+rpm -ihv $(ls -ltr yggdrasil-*.src.rpm | tail -n 1 | awk '{print $NF}')
+if [ "$ARCH" = "aarc64" ]; then
+    # Turn ELF binary stripping off in %post
+    sed -i '1s/^/%global __os_install_post %{nil}/' ~/rpmbuild/SPECS/yggdrasil.spec
+fi
+rpmbuild -bb ~/rpmbuild/SPECS/yggdrasil.spec --target $ARCH
+
+# Build k4e-device-worker rpm
+git clone https://github.com/jakub-dzon/k4e-device-worker.git /home/builder/k4e-device-worker
+cd /home/builder/k4e-device-worker
+if [ "$ARCH" = "aarc64" ]; then
+    make build-arm64
+    make rpm-arm64
+else
+    make build
+    make rpm
+fi
+
+#---------------------------
+# Create packages repository
+#---------------------------
+dnf install -y createrepo httpd
+systemctl enable --now httpd
+firewall-cmd --zone=public --add-port=80/tcp --permanent
+firewall-cmd --reload
+
+mkdir /var/www/html/k4e-repo
+cp /root/rpmbuild/RPMS/${ARCH}/*.${ARCH}.rpm /var/www/html/k4e-repo/
+createrepo /var/www/html/k4e-repo/
+
+#-----------------------------------
+# Create and publish rhel4edge image
+#-----------------------------------
+git clone https://github.com/ydayagi/r4e.git /home/builder/r4e
+cd /home/builder/r4e
+./r4e-image.sh $(hostname) edgedevice helios05.lab.eng.tlv2.redhat.com:8888 http://$(hostname)/k4e-repo/
+
+#----------------------------------------
+# Create ISO for image and serve via http
+#----------------------------------------
+cat << EOF > /tmp/empty.toml
```

Comment:
this is the new way for creating ISO?

Reply:
yes.
one thing that I need to verify about the created ISO is the content of ks file in it.

Reply:
it's not fetching the ks file we create?

Reply:
Based on this [info](https://github.com/redhat-cop/rhel-edge-automation-arch/blob/main/docs/basic-walkthrough.md#creating-auto-booting-rfe-installer) from automation repo kickstart file is embedded in to the ISO.

Reply:
The content of the created ISO file has this file in it:
```
cat osbuild.ks 
ostreesetup --osname=rhel --url=file:///run/install/repo/ostree/repo --ref=rhel/8/x86_64/edge --nogpg
```

this file is being referred from the other files that are involved in the boot sequence:
```
grep -Rl  osbuild.ks
EFI/BOOT/BOOT.cfg
EFI/BOOT/grub.cfg
images/efiboot.img
isolinux/grub.conf
isolinux/isolinux.cfg
```

It is my understanding that we need to edit `osbuild.ks` and not `isolinux/isolinux.cfg`.
Also, `isolinux/isolinux.cfg` is available only for x86_64 and not for aarch64.

Reply:
Additional info about the file: https://github.com/achilleas-k/osbuild-composer/commit/579a0765fa6e4c56296aa1aaaf5d0893c442cfa3

Reply:
so editing the kickstart is not part of this PR? Is the ISO usable this way? Cause in the current iso we create there are post install steps in ks file.